scalameter-core/src/main/scala/org/scalameter/execution/invocation/instrumentation/Instrumentation.scala

```diff
@@ -0,0 +1,107 @@
+package org.scalameter.execution.invocation.instrumentation
+
+import java.io._
+import java.util.jar.{JarEntry, JarOutputStream, Attributes}
+import java.util.zip.ZipFile
+import org.objectweb.asm.{ClassReader, ClassWriter}
+import scala.collection.convert.decorateAsScala._
+import org.scalameter._
+import org.scalameter.execution.invocation._
+
+
+private[scalameter] object Instrumentation {
+  /** Reads classes from given location and transforms them if they match a predicate.
+   *
+   *  Note that it supports reading from directories and jars.
+   *
+   *  @param in location of classes to read from
+   *  @param p matches if class should be instrumented
+   *  @param instrumentFrom instruments class from a [[java.io.InputStream]]
```

Comment:
"a function that instruments a given class represented by its classname (?) and the input stream of its contents, and outputs the instrumented class to the resulting byte array"